### PR TITLE
chore: Embed sample configurations into README for inputs

### PR DESCRIPTION
**Pull request by srebhan**
- [x] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

This PR uses the newly created `sample.conf` files as a single source of truth for input plugins. To do so, the `sample.conf` is embedded into the plugin package at build time using [Golang's embed package](https://pkg.go.dev/embed). Furthermore, the `sample.conf` file is _input_ into the `REAME.md` markdown file by using a custom `@<file to input>` pattern then used by the new `readme_config_includer/generator` tool to do the heavy lifting.
This new approach will remove the current stateful build (for inputs) and improves build times.

For validating the approach, the generated telegraf sample-configuration was compared before and after the PR and is binary equal. Furthermore, the current README.md files can be reproduced. You can test this by removing the content of a `toml @sample.conf` section in a README.md and run `make`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -115,13 +115,22 @@ versioninfo:
...
+
 .PHONY: generate
-generate:
-	go generate -run="plugindata/main.go$$" ./plugins/inputs/... ./plugins/outputs/... ./plugins/processors/... ./plugins/aggregators/...
+generate: insert_config_to_readme_inputs generate_plugins_outputs generate_plugins_processors generate_plugins_aggregators
 
 .PHONY: generate-clean
 generate-clean:
-	go generate -run="plugindata/main.go --clean" ./plugins/inputs/... ./plugins/outputs/... ./plugins/processors/... ./plugins/aggregators/...
+	go generate -run="plugindata/main.go --clean" ./plugins/outputs/... ./plugins/processors/... ./plugins/aggregators/...
```
Comment (sspaink) [inline Makefile:133]: Is the `generate-clean command` still used anywhere? Should it be deleted?

Comment (srebhan) [reply Makefile:133]: Currently it is useful for `outputs`, `processors` and `aggregators`. I will remove it when everything is migrated.

File: tools/readme_config_includer/generator.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+			return errors.New("adding newline failed")
+		}
+	}
+
+	return nil
+}
+
+func main() {
+	goPackage := os.Getenv("GOPACKAGE")
+	_ = goPackage
```
Comment (sspaink) [inline tools/readme_config_includer/generator.go:78]: Why do we need to do this? I don't see `goPackage` being used anywhere.

Comment (srebhan) [reply tools/readme_config_includer/generator.go:78]: Good catch! Seems to be a left-over... _Note to myself: DON'T DO THIS!_

Comment (srebhan) [reply tools/readme_config_includer/generator.go]: Fixed.

**Review by sspaink — APPROVED** [review]
So I didn't look at all 675 files, but I did review the `readme_config_includer` tool and I think it looks good. I tried it locally as well and it seems to be working. Thank you for all this work!

**Comment by srebhan** [comment]
Rebased with no changes to resolve conflict.

**Comment by srebhan** [comment]
Rebase to solve merge conflict.

**Pull request merged**